src/RamTransforms.cpp

```diff
@@ -35,9 +35,9 @@ namespace {
 std::vector<std::unique_ptr<RamCondition>> getConditions(const RamCondition* condition) {
     std::vector<std::unique_ptr<RamCondition>> conditions;
     while (condition != nullptr) {
-        if (const auto* ramAnd = dynamic_cast<const RamConjunction*>(condition)) {
-            conditions.emplace_back(ramAnd->getRHS().clone());
-            condition = &ramAnd->getLHS();
+        if (const auto* ramConj = dynamic_cast<const RamConjunction*>(condition)) {
+            conditions.push_back(std::unique_ptr<RamCondition>(ramConj->getRHS().clone()));
```

Comment:
RamCondition into vector of unique points of RamCondition
```suggestion
            conditions.emplace_back(ramConj->getRHS().clone());
```